unit-tests/dds/test-stream-sensor-bridge.py

```diff
@@ -0,0 +1,341 @@
+finally:
+    reset()
+test.finish()
+#
+#############################################################################################
+#
+test.start( "incompatible streams" )
+try:
+    bridge.open( servers['Infrared 1'].default_profile() )  # 1280x800 GREY @ 30 Hz
+    bridge.add_implicit_profiles()  # IR2
+    test.check_equal( len(active_sensors), 0 )  # not streaming yet
+    with bridge_error_expected( "failure trying to start/stop 'Depth': profile <'Depth' 1280x720 Z16 @ 30 Hz> is incompatible with already-open <'Infrared 1' 1280x800 GREY @ 30 Hz>" ):
+        start_stream( 'Depth' )  # no depth at 1280x800, so no stream!
+    test.check_equal( len(active_sensors), 0 )
+    test.check_throws( lambda: 
+        bridge.open( servers['Depth'].default_profile() ),
+        RuntimeError, "profile <'Depth' 1280x720 Z16 @ 30 Hz> is incompatible with already-open <'Infrared 1' 1280x800 GREY @ 30 Hz>" )
+    test.check_throws( lambda: 
+        bridge.open( find_server_profile( 'Infrared 2', '848x480 GREY @ 30 Hz' )),
+        RuntimeError, "profile <'Infrared 2' 848x480 GREY @ 30 Hz> is incompatible with already-open <'Infrared 1' 1280x800 GREY @ 30 Hz>" )
+    test.check_throws( lambda: 
+        bridge.open( find_server_profile( 'Infrared 2', '1280x800 GREY @ 15 Hz' )),
+        RuntimeError, "profile <'Infrared 2' 1280x800 GREY @ 15 Hz> is incompatible with already-open <'Infrared 1' 1280x800 GREY @ 30 Hz>" )
+    start_stream( 'Infrared 2' )
+    ( test.check_equal( len(active_sensors), 1 )
+        and test.check_equal( next(iter(active_sensors)), 'Stereo Module' )
+        and test.check_equal( len(active_sensors['Stereo Module']), 2 )  # IR1, IR2
+        )
+    test.check_throws( lambda: 
+        bridge.open( servers['Depth'].default_profile() ),
+        RuntimeError, "sensor 'Stereo Module' was committed and cannot be changed" )
+except Exception:
+    test.unexpected_exception()
+finally:
+    reset()
+test.finish()
+#
+#############################################################################################
+#
+test.start( "open and close" )
+try:
+    bridge.open( servers['Infrared 1'].default_profile() )  # 1280x800 GREY @ 30 Hz
+    bridge.open( servers['Infrared 1'].default_profile() )  # "compatible"
+    bridge.close( servers['Infrared 1'] )
+    bridge.open( find_server_profile( 'Infrared 1', '1280x800 GREY @ 30 Hz' ))
+    bridge.close( servers['Infrared 1'] )
+    bridge.close( servers['Infrared 1'] )
+    bridge.open( find_server_profile( 'Infrared 1', '1280x800 Y16 @ 25 Hz' ))
+    bridge.reset()
+    bridge.open( find_server_profile( 'Infrared 1', '1280x800 Y16 @ 15 Hz' ))
+    test.check_throws( lambda:
+        bridge.open( find_server_profile( 'Infrared 1', '1280x800 Y16 @ 25 Hz' )),
+        RuntimeError, "profile <'Infrared 1' 1280x800 Y16 @ 25 Hz> is incompatible with already-open <'Infrared 1' 1280x800 Y16 @ 15 Hz>" )
+except Exception:
+    test.unexpected_exception()
+finally:
+    reset()
+test.finish()
```

Comment:
Should also test motion profiles

Reply:
That doesn't exactly test bridge functionality, but OK I'll add one. Any specific scenario you have in mind?

Reply:
Only one motion.
Gyro+accel (like depth+ IR1)
One motion + One video. Stop one of them see that the other still streams

Reply:
Done. Check out the comments I added in the test.

Reply:
I do think that the device needs to catch errors, especially with streaming. But not in this PR.

Reply:
Didn't understand the comment about the FPS.

Reply:
> Done. Check out the comments I added in the test.
You had a comment `#Gyro has no 400 Hz`. My D435i does have 400Hz.

Reply:
But not the D435i in the test, for some reason...